src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -2442,5 +2445,137 @@ private void InterpretLoadElement(int token)
                     throw new NotImplementedException();
             }
         }
+
+        private void InterpretCall(int token)
+        {
+            MethodDesc method = (MethodDesc)_methodIL.GetObject(token);
+            MethodSignature signature = method.Signature;
+
+            TypeDesc owningType = method.OwningType;
+            TypeDesc returnType = signature.ReturnType;
+
+            if (signature.Length > _stack.Count)
+                ThrowHelper.ThrowInvalidProgramException();
+
+            int delta = (signature.IsStatic ? 1 : 2);
+
+            LocalVariableType[] localVariableTypes = new LocalVariableType[signature.Length + delta];
+            localVariableTypes[0] = new LocalVariableType(returnType.GetRuntimeTypeHandle(), false, returnType.IsByRef);
+
+            if (!signature.IsStatic)
+                localVariableTypes[1] = new LocalVariableType(owningType.GetRuntimeTypeHandle(), false, owningType.IsByRef);
+
+            for (int i = 0; i < signature.Length; i++)
+            {
+                var argument = signature[i];
+                localVariableTypes[i + delta] = new LocalVariableType(argument.GetRuntimeTypeHandle(), false, argument.IsByRef);
+            }
+
+            int neededMemory = LocalVariableSet.ComputeNecessaryMemoryForStackLocalVariableSet(localVariableTypes);
+            IntPtr* pbMemory = stackalloc IntPtr[neededMemory];
+            LocalVariableSet localVariableSet = new LocalVariableSet(pbMemory, localVariableTypes);
+            LocalVariableSet.DefaultInitializeLocalVariableSet(ref localVariableSet);
```

Comment:
I think the problem is that I may not be setting up the `LocalVariableSet` properly

Reply:
We should allocate `LocalVariableSet` through `LocalVariableSet.SetupArbitraryLocalVariableSet`.

The flow should be something like:
* Allocate an array of `LocalVariableType` that represents the types in the signature (you already have that part)
* Call `LocalVariableSet.SetupArbitraryLocalVariableSet`, passing in the `LocalVariableType` array, a delegate to a callback that sets up the parameters, and a helper struct (the helper struct is something you can define here - the helper struct will probably have the address of the method entry point, parameter values, calling convention, return value, and the like).
* `LocalVariableSet.SetupArbitraryLocalVariableSet` will call your callback method with an already allocated `LocalVariableSet` - this is where you'll do the `SetVar` calls, call `CallInterceptor.MakeDynamicCall`, and stash away the return value in the helper struct.
* ...
* Profit

Reply:
I think my biggest confusion with `SetupArbitraryLocalVariableSet` is what the generic `T` parameter is supposed to represent

Reply:
Is it the helper struct?